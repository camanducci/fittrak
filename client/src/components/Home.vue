<template>
<v-layout justify-center row wrap>
  <StartWorkout></StartWorkout>
  <LatestWorkouts></LatestWorkouts>
</v-layout>
</template>

<script>
import moment from 'moment'
import StartWorkout from './workouts/StartWorkout'
import LatestWorkouts from './workouts/LatestWorkouts'

export default {
  data: () => ({
    today: moment().format('MMMM Do, YYYY')
  }),

  components: {
    StartWorkout,
    LatestWorkouts
  }
}
</script>

