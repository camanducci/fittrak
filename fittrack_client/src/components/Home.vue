<template>
  <div>
    <h2> It's {{ today }} </h2>
    <router-link to="/startworkout" class="button is-large is-primary is-outlined">New Workout</router-link>
  </div>

</template>

<style>
</style>

<script>
import moment from 'moment'

export default {
  data () {
    return {
      today: moment().format('MMMM Do, YYYY')
    }
  }
}
</script>

