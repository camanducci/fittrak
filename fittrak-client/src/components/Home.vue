<template>
<v-layout justify-center row wrap>
  <WorkoutActions></WorkoutActions>
  <RecentWorkouts></RecentWorkouts>
</v-layout>
</template>

<script>
import moment from 'moment'
import WorkoutActions from './workouts/WorkoutActions'
import RecentWorkouts from './workouts/RecentWorkouts'

export default {
  data: () => ({
    today: moment().format('MMMM Do, YYYY')
  }),

  components: {
    WorkoutActions,
    RecentWorkouts
  }
}
</script>
