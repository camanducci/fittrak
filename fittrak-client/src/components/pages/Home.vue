<template>
  <div>
    <WorkoutList title="Current Workouts" />
    <CreateWorkout />
  </div>
</template>

<script>
import CreateWorkout from "@/components/workouts/CreateWorkout.vue";
import WorkoutList from "@/components/workouts/WorkoutList.vue";

export default {
  name: "home",
  components: {
    CreateWorkout,
    WorkoutList
  }
};
</script>
