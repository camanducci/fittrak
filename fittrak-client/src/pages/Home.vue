<template>
  <div class="page-content">
    <h1 class="title">It's {{ now }}! Time to get crackin'!</h1>

    <h2 class="page-header">Ongoing Workouts</h2>

    <hr />
    <CreateWorkout />
    <WorkoutList :status="status" />
  </div>
</template>

<script>
import CreateWorkout from "@/components/workouts/CreateWorkout.vue";
import WorkoutList from "@/components/workouts/WorkoutList.vue";

import { PENDING } from "@/constants";

import { format } from "date-fns";

export default {
  name: "home",
  computed: {
    now() {
      const formatted = format(new Date(), "dddd");

      return formatted;
    },
    status() {
      return PENDING;
    }
  },
  components: {
    CreateWorkout,
    WorkoutList
  }
};
</script>
