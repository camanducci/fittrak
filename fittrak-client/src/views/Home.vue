<template>
  <div>
    <WorkoutList />
  </div>
</template>

<script>
import WorkoutList from "@/components/WorkoutList.vue";

export default {
  name: "home",
  components: {
    WorkoutList
  }
};
</script>
