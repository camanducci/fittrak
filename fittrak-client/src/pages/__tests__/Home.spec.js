import { shallowMount } from "@vue/test-utils";

import Home from "@/pages/Home";

describe("Home", () => {
  test("Home page renders", () => {
    const app = shallowMount(Home);
    expect(app.html()).toMatchSnapshot();
  });
});
