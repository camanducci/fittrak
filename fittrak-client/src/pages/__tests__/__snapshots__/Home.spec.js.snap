// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home page renders 1`] = `
<v-container-stub tag="div" fluid="" grid-list-xl="">
  <v-layout-stub tag="div" row="">
    <v-flex-stub tag="div">
      <workoutfilterheader-stub></workoutfilterheader-stub>
    </v-flex-stub>
  </v-layout-stub>
  <v-layout-stub tag="div" fill-height="">
    <v-flex-stub tag="div">
      <workoutcardlist-stub workouts=""></workoutcardlist-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
